docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,331 @@
+
+% made CERT-3797 so we get a clear indication to the user, hopefully
+% then, CVT should indicate via report-logs
+%  - whether there was a hard stop
+%  - whether SMT has started
+
+
+Type 3. is signified by a soft stop. This means an SMT solver shut down due to
+hitting the limit for a single SMT run (set via `--smt_timeout`). When running
+with default settings this means that we give up for the individual rule since
+in order to obtain a proof of correctness we need to solve every subproblem we
+generate (see documentation on [control-flow splitting](control-flow-splitting)
+for more details). 
+
+In the remainder we will focus on the mitigation of SMT timeouts, i.e., types 2.
+and 3. Non-SMT Timeouts (Type 1.) should be reported to Certora. 
+
+(timeout_causes)=
+# What Causes Timeouts?
+
+As a first step towards resolving a timeout, we need to diagnose its root
+causes. In our experience so far, the following are some of the most common
+reasons for SMT timeouts:
+
+ - Non-trivial amount of nonlinear arithmetic
+ - Very high path count
+ - High storage/memory complexity
+
+This list is not exhaustive, but the majority of timeouts we have observed so
+far can be traced back to one or more of these causes. While these are not the
+only sources of complexity, they should work well as general estimates of
+complexity. For instance, linear arithmetic usually only becomes a problem when
+the input program is rather large, which is also indicated by path count in most
+practical cases.
+
+
+## Complexity Feedback from Certora Prover
+
+Certora Prover provides some help with diagnosing timeouts. We present these
+features in this section.
+
+### Difficulty Statistics
+
+Certora Prover provides statistics on the problem sizes it encounters. 
+These statistics are structured according to the timeout reasons given above.
+
+Currently, the Prover tracks the following statistics:
+ - Path count
+ - Nonlinear operations count
+ - Memory/storage complexity
+
+For a very short summary we give one summarizing number for each of the
+statistics, along with a LOW/MEDIUM/HIGH statement. This occurs as an INFO
+message in the Global Problems pane of the Prover reports.
+
+The exact classifications are made from experience with these statistics.
+ - LOW: unlikely to be a reason for a timeout
+ - MEDIUM: might be a reason for a timeout; the timeout might also be a result
+   of the combined complexity with other measures
+ - HIGH: likely to be a reason for a timeout, even if it is the only aspect of
+   the verification problem that shows high complexity
+
+These categories map to intervals as follows.
+
+|    | LOW | MEDIUM | HIGH |
+|----|-----|--------|------|
+| Path count | 0 to 2<sup>20</sup> | 20<sup>20</sup> to 2<sup>80</sup> | > 2<sup>80</sup> |
+| Nonlinear operations | 0 to 10 | 10 to 30 | > 30 |
+
+```{note}
+All heuristics mentioned here are might be improved over time, so 
+consider them subject to change.
```

Comment:
Actually, I would remove the whole `{note}`. Everything CVT related might be improved over time and henced considered subject to change.

Reply:
alright .. I'm a bit over-cautious at times ..